### ISSUE TITLE
Random number generation in some analysis functions

### ISSUE THREAD

**Issue by crisluengo** [DIPlib/diplib#47]
~~In DIPimage, `private/dip_generation.cpp` has a `dip::Random` object that is default-initialized when the MEX-file is loaded, and cannot be controlled by the user. It is important to be able to set the seed. We could add a function to do so here. **But:** in the future, when adding functions that need a dip::Random object, they will need to be added to this MEX-file.~~

~~PyDIP has the same issue: the `dip::Random` object is created and used in `assorted.cpp`.~~

---

There are a few functions that create their own random object when called:

- `dip::ChordLength`
- `dip::PairCorrelation`
- `dip::ProbabilisticPairCorrelation`
- `dip::Semivariogram`
- `dip::CostesSignificanceTest`
- `dip::StochasticWatershed`
- `dip::KMeansClustering`
- `dip::Superpixels`

These functions do not allow controlling of the randomness. Does this need to change? The problem with these is that you cannot use a default value for the `dip::Random` object, if you want to take the object by reference. Thus, by allowing the user to pass in a random number generator object, you are forcing the user to pass one in. If there was a way of providing a default one, this would be an easy choice!

**Comment by crisluengo** [comment]
Main issue fixed in PyDIP in 846127d9 and in DIPimage in c2cc1bd5.

Leaving this open for now to think a bit more about the use of default-initialized RNGs inside some analysis functions.

**Comment by crisluengo** [comment]
The way to provide a default is to create a 2nd interface to the function:

```cpp
// Base function, no default values before the `random` argument.
void Function(type1 arg1, type2 arg2, Random& random);

// Alternate interface, which does not have a `random` argument.
// Here we can provide default values for other arguments.
void Function(type1 arg1, type2 arg2=0) {
   Random random; // default initialized, which is totally fine.
   Function(arg1, arg2, random);
}
```
